File: VB/NavBarSampleSL/CustomNavigationPaneView.vb
Imports DevExpress.Xpf.NavBar

Namespace NavBarSampleSL
    Public Class CustomNavigationPaneView
        Inherits NavigationPaneView

        Public Property OpenedWidth() As Double
        Public Property PreviousWidth() As Double
        Protected Overrides Sub UpdatePresenterTemplateByEnabledState()
            If OpenedWidth <> 0 Then
                ExpandedWidth = OpenedWidth
            End If
            MyBase.UpdatePresenterTemplateByEnabledState()
        End Sub
    End Class
End Namespace

File: CS/NavBarSampleSL/CustomNavigationPaneView.cs
using DevExpress.Xpf.NavBar;

namespace NavBarSampleSL {
    public class CustomNavigationPaneView : NavigationPaneView {
        public double OpenedWidth { get; set; }
        public double PreviousWidth { get; set; }
        protected override void UpdatePresenterTemplateByEnabledState() {
            if (OpenedWidth != 0) {
                ExpandedWidth = OpenedWidth;
            }
            base.UpdatePresenterTemplateByEnabledState();
        }
    }
}
